wg-api/spec-documents/scroll-view.md

```diff
@@ -0,0 +1,259 @@
+scrollContent.addBrowserView(wrapperBrowserView2)
+
+scroll.setContentView(scrollContent);
+win.addChildView(scroll);
+```
+
+
+## API Design
+
+In order to achieve the example above, we've made following changes to the API
+
+## BaseView
+
+A `BaseView` is a rectangle within the views View hierarchy. It is the base
+class for all all different views. 
+
+Process: **Main**
+
+`BaseView` is an EventEmitter
+
+### Static Methods
+
+#### `BaseView.getAllViews()`
+
+Returns the array of all created views.
+
+#### `BaseView.fromId(id)`
+
+Returns `BaseView | null` - The view with the given `id`.
+
+### Instance Properties
+
+#### `view.id`
+
+A `Integer` property representing the unique ID of the view.
+
+#### `view.isContainer`
+
+A `boolean` property that determines whether this view is `ContainerView`.
+
+### Instance Methods
+
+#### `view.setBounds(bounds)`
+
+Resizes and moves the view to the supplied bounds relative to its parent.
+
+#### `view.getBounds()`
+
+Returns the position and size of the view.
+
+#### `view.setBackgroundColor(color)`
+
+Change the background color of the view.
+
+#### `view.getParentView()`
+
+Returns `BaseView || null` - The parent view, otherwise returns `null`.
+
+#### `view.getParentWindow()`
+
+Returns `BrowserWindow || null` - The window that the view belongs to, otherwise returns `null`.
+
+### Events
+
+#### Event: 'size-changed'
+
+Emitted when the view's size has been changed.
+
+ 
+## BrowserWindow 
+
+### New instance methods
+
+#### `win.addChildView(view)`
+
+#### `win.removeChildView(view)`
+
+#### `win.getViews()`
+
+Returns `BaseView[]` - an array of all BaseViews that have been attached
+with `addChildView`.
+
+
+## ContainerView
+
+A `ContainerView` can be used to embed additional views hierarchy into a `BrowserWindow`. It extends `BaseView`.
+
+Process: **Main**
+
+### Instance Methods
+
+#### `containerView.addChildView(view)`
+
+#### `containerView.removeChildView(view)`
+
+#### `containerView.getViews()`
+
+Returns `BaseView[]` - an array of all BaseViews that have been attached
+with `addChildView`.
+
+
+## ScrollView
+
+Show a part of view with scrollbar. 
+The `ScrollView` can show an arbitrary content view inside it. It is used to make
+any View scrollable. When the content is larger than the `ScrollView`,
+scrollbars will be optionally showed. When the content view is smaller
+then the `ScrollView`, the content view will be resized to the size of the
+`ScrollView`.
+The scrollview supports keyboard UI and mousewheel.
+It extends `BaseView`.
+
+Process: **Main**
+
+### Static Methods
+
+#### `ScrollView.fromId(id)`
+
+Returns `ScrollView | null` - The scroll view with the given `id`.
+
+### Instance Methods
+
+#### `scrollView.setContentView(contents)`
+
+Set the contents. The contents is the view that needs to scroll.
+
+#### `scrollView.getContentView()`
+
+Returns `BaseView` - The contents of the `scrollView`.
+
+#### `scrollView.setContentSize(size)`
```

Comment:
it seems a bit odd to me to set the size of the child view through a method on the parent. which way does this information flow? Does the content view choose its own size, or does the scroll view set the size of the content?

Reply:
On Mac I defined this function as follows:
```
void NativeScrollView::SetContentSize(const gfx::Size& size) {
  auto* scroll = static_cast<ElectronNativeScrollView*>(GetNative());
  NSSize content_size = size.ToCGSize();
  [scroll setContentSize:content_size];
  [scroll.documentView setFrameSize:content_size];
}
```
That's the only way to set the content's size I think. Using setBounds for content of the scrollView causes the issue when scrollView's bounds are changed (scroll's content stops working).